### PR TITLE
[MP Rewrite Part 7] Add global obstacles; fix dynamic obstacle properties

### PR DESCRIPTION
**Pull request by kylestach**
This PR adds global and goal-zone obstacles to the planning stack. It also fixes how we deal with dynamic obstacles by essentially treating them as static after they reach the end of their trajectory.

### CODE REVIEW

File: soccer/gameplay/GameplayModule.cpp [hunk 1]
```diff
@@ -314,6 +314,9 @@ void Gameplay::GameplayModule::run() {
     _ballMatrix = Geometry2d::TransformMatrix::translate(
         _context->world_state.ball.position);
 
+    _context->globalObstacles = globalObstacles();
+    _context->goalZoneObstacles = goalZoneObstacles();
```
Comment (JNeiger) [inline soccer/gameplay/GameplayModule.cpp:281]: Does it make sense to have the global obstacles set here? It doesn't really feel like it fits. For the sake of this PR though, this is fine

**Review by JNeiger — APPROVED** [review]

File: soccer/planning/low_level/CreatePath.cpp [hunk 2]
```diff
@@ -52,7 +52,7 @@ Trajectory rrt(const LinearMotionInstant& start,
 
     ShapeSet obstacles = static_obstacles;
     Trajectory path{{}};
-    constexpr int attemptsToAvoidDynamics = 10;
+    constexpr int attemptsToAvoidDynamics = 3;
```
Comment (oswinso) [inline soccer/planning/low_level/CreatePath.cpp]: I guess probably make this a param as well?

**Review by oswinso — APPROVED** [review]
lgtm

File: soccer/gameplay/GameplayModule.cpp (hunk 1 above)
Comment (kylestach) [reply soccer/gameplay/GameplayModule.cpp:281]: Yeah this has always been where they're created but I think it really belongs with the field geometry calculations.

**Pull request merged**